integration/auth/user_mgt_test.go

```diff
@@ -34,16 +36,26 @@ var testFixtures = struct {
 }{}
 
 func TestUserManagement(t *testing.T) {
-	t.Run("Create test users", testCreateUsers)
-	t.Run("Get user", testGetUser)
-	t.Run("Iterate users", testUserIterator)
-	t.Run("Paged iteration", testPager)
-	t.Run("Disable user account", testDisableUser)
-	t.Run("Update user", testUpdateUser)
-	t.Run("Remove user attributes", testRemovePhonePhotoName)
-	t.Run("Remove custom claims", testRemoveCustomClaims)
-	t.Run("Add custom claims", testAddCustomClaims)
-	t.Run("Delete test users", testDeleteUsers)
+	orderedRuns := []struct {
+		name     string
+		testFunc func(*testing.T)
+	}{
+		{"Create test users", testCreateUsers},
+		{"Get user", testGetUser},
+		{"Iterate users", testUserIterator},
+		{"Paged iteration", testPager},
+		{"Disable user account", testDisableUser},
+		{"Update user", testUpdateUser},
+		{"Remove user attributes", testRemovePhonePhotoName},
+		{"Remove custom claims", testRemoveCustomClaims},
+		{"Add custom claims", testAddCustomClaims},
+		{"Delete test users", testDeleteUsers},
+	}
+	for _, run := range orderedRuns {
+		if ok := t.Run(run.name, run.testFunc); !ok {
+			t.Fatalf("Failed run %v", run.name)
```

Comment:
Why is this needed? Should add a comment if it's intentional

Reply:
Adding comment. 
	// The tests are meant to be run in sequence, a failure in creating the users
	// should be fatal so non of the other tests run. However calling Fatal from a
	// subtest does not prevent the other subtests from running, hence we check the
	// success of each subtest before proceeding.